bql/semantic/expression.go

```diff
@@ -197,6 +153,33 @@ func (e *comparisonForLiteral) Evaluate(r table.Row) (bool, error) {
 	}
 }
 
+// comparisonForNodeLiteral represents the internal representation of an expression of comparison between a binding and a node literal.
+type comparisonForNodeLiteral struct {
+	op OP
```

Comment:
I wonder if we should add comments to the fields of these structs.
It is kind of hard to figure out what each field means without seeing them in being used in the code.

Reply:
That is true, it is indeed, I have just added more comments for them.